Add route for vote page

androidgrl opened the pull request.

androidgrl committed:
Add route for vote page

stevekinney commented on lib/poll.js:
```diff
@@ -3,8 +3,8 @@ const crypto = require('crypto');
-  this.adminUrl = `http://localhost:3000/${this.id}`;
-  this.voterUrl = `http://localhost:3000/${this.voterId}`;
+  this.adminUrl = `http://localhost:3000/admin/${this.id}`;
+  this.voterUrl = `http://localhost:3000/vote/${this.voterId}`;
```
We probably don't want to hard-code in `http://localhost:3000`. These will break in production. Also, they're not super dry. If for some reason down the road `this.id` or `this.voterId` changed (Imagine the user sent out the wrong one and wanted to reset their `id`—this isn't super uncommon), then you'd have to change this in two places.

Alternatively, you could use methods here:

``` js
Poll.prototype.adminUrl = function () {
  return '/admin/' + this.id;
}
```

stevekinney commented on lib/poll.js:
```diff
@@ -3,8 +3,8 @@ const crypto = require('crypto');
 function Poll() {
   this.id = crypto.randomBytes(20).toString('hex');
   this.voterId = crypto.randomBytes(20).toString('hex');
```
We're using `crypto.randomBytes(20).toString('hex')` multiple times. It probably makes sense to extract it out into a helper function.

Something along the lines of:

``` js
function generateHash() {
  return crypto.randomBytes(20).toString('hex')
}
```

androidgrl committed:
Add redis

Sebastian showed me how to do this, it will
save a stringified version of our poll object.

androidgrl committed:
Render the poll in the vote view

To have access to the object we had to
render it inside the redis request.

Tess helped us with this, the problem is you
can't access the object outside of the request.

androidgrl committed:
Dynamically set the host

So that we can easily switch between local host
and heroku.

androidgrl committed:
Extract generate hash to its own function

imwithsam commented:
:100:
